scripts/ci/jenkins_diff.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/sh -le
+
+# This is an Jenkins-specific script that runs infracost on the current branch.
+# Usage docs: https://www.infracost.io/docs/integrations/
+
+fix_env_vars () {
+    # Jenkins has problems with envs case sensivity
+    iac_path=${iac_path:-$IAC_PATH}
+    terraform_plan_flags=${terraform_plan_flags:-$TERRAFORM_PLAN_FLAGS}
+    terraform_workspace=${terraform_workspace:-$TERRAFORM_WORKSPACE}
+    usage_file=${usage_file:-$USAGE_FILE}
+    config_file=${config_file:-$CONFIG_FILE}
+    fail_condition=${fail_condition:-$FAIL_CONDITION}
+}
+
+process_args () {
+  # Set variables based on the order for GitHub Actions, or the env value for other CIs
+  iac_path=${1:-$iac_path}
+  terraform_plan_flags=${2:-$terraform_plan_flags}
+  terraform_workspace=${3:-$terraform_workspace}
+  usage_file=${4:-$usage_file}
+  config_file=${5:-$config_file}
+  fail_condition=${7:-$fail_condition}
+
+  # Validate fail_condition
+  if ! echo "$fail_condition" | jq empty; then
+    echo "Error: fail_condition contains invalid JSON"
+  fi
+
+  # Set defaults
+  if [ ! -z "$fail_condition" ] && [ "$(echo "$fail_condition" | jq '.percentage_threshold')" != "null" ]; then
+    fail_percentage_threshold=$(echo "$fail_condition" | jq -r '.percentage_threshold')
+  fi
+  fail_percentage_threshold=${fail_percentage_threshold:--1}
```

Comment:
hmm I wasn't expecting this to default to -1 since for [other CI integrations](https://github.com/infracost/infracost-atlantis/#post_condition) this is the absolute percentage threshold:
"For example, set to 1 to post a comment if the cost estimate changes by more than plus or minus 1%."

Could we do something similar here? I'm actually not sure how useful the drops in cost posting a comment or failing the build are but I lean towards keeping thins consistent until we have more feedback 🤷‍♂️

Reply:
Oh I see, I actually had a different intention. Since setting the default value to 0 will collide with "Fail if there was any change in diff (setting the threshold to 0)", I set its default value as -1. Do you have any suggestions to enhance it?

Reply:
So we'd want to handle 2 cases:
1. never fail the build. Default behavior. Maybe we can do that by checking if `fail_percentage_threshold` is **not present**?
2. fail the build if the costs increase **or** decrease by X%, can be done using the fail_percentage_threshold

Reply:
Makes sense, done.